### DISCUSSION TITLE
Best way to work on a map in a team

### DISCUSSION THREAD

**Discussion by Bertik23** [OpenOrienteering/mapper#1934] [Questions & Answers]
Hey, I'll be working on a map with my friend and I'm searching for the best way to do so. I have already done some tests with git and Github, but that won't work as everytime a repo is a commit behind another a merge conflict occures when trying to pull. That wouldn't be a issue for me. As I'm used to git, but my friend isn't and it could be quite frustrating to use this way.

The only other sensible option I found was to make all changes on a new map part and than when syncing up with each other we would just delete everything except the new map part and save it as a new map and send it. And import it on the other end.

Is there a way to fix the merge conflicts or a much easier way to making a new map part and importing it?

**Comment by dg0yt** [comment]
> The only other sensible option I found was to make all changes on a new map part and than when syncing up with each other we would just delete everything except the new map part and save it as a new map and send it. And import it on the other end.

I think this is the best option at the moment. The map parts make this more feasible, but I acknowledge that this is still not as convenient as desired.

There is already some discussion, including recent comments, on https://github.com/OpenOrienteering/mapper/issues/1290. If there is sufficient interest in testing changes, i.e. *user-side commitment to testing master snapshots*, I might implement incremental improvements.

**Reply by Bertik23** [reply]
Maybe adding the possibility to export a map part as a new map would be easier and provide a quick fix while you are working on a better solution.

Not sure how doable that is tho. I tried looking at the code but I have zero experience with C++ so I got lost very quickly.

**Reply by dg0yt** [reply]
> Maybe adding the possibility to export a map part as a new map would be easier and provide a quick fix while you are working on a better solution.

And a few more steps:
- Add a template map there, made from the original minus that part.
- Add locking of colors and symbols. 
- Last not least, a "Replace map part" action  for re-import.

**Reply by ghost** [reply]
Also, here is additional discussion on collaborative mapping via map parts & importing shared files each to another:

- https://github.com/OpenOrienteering/mapper/discussions/1886
